ambry-cloud/src/main/java/com/github/ambry/cloud/azure/CosmosDataAccessor.java

```diff
@@ -144,47 +141,37 @@ void testConnectivity() {
   ResourceResponse<Document> deleteMetadata(CloudBlobMetadata blobMetadata) throws DocumentClientException {
     String docLink = getDocumentLink(blobMetadata.getId());
     RequestOptions options = getRequestOptions(blobMetadata.getPartitionId());
-    options.setPartitionKey(new PartitionKey(blobMetadata.getPartitionId()));
     return executeCosmosAction(() -> asyncDocumentClient.deleteDocument(docLink, options).toBlocking().single(),
         azureMetrics.documentDeleteTime);
   }
 
   /**
-   * Read the blob metadata document in the CosmosDB collection.
-   * @param blobId the {@link BlobId} for which metadata is requested.
-   * @return the {@link ResourceResponse} containing the metadata document.
-   * @throws DocumentClientException if the operation failed.
-   */
-  ResourceResponse<Document> readMetadata(BlobId blobId) throws DocumentClientException {
-    String docLink = getDocumentLink(blobId.getID());
-    RequestOptions options = getRequestOptions(blobId.getPartition().toPathString());
-    return executeCosmosAction(() -> asyncDocumentClient.readDocument(docLink, options).toBlocking().single(),
-        azureMetrics.documentReadTime);
-  }
-
-  /**
-   * Update the blob metadata document in the CosmosDB collection, retrying as necessary.
+   * Update the blob metadata document in the CosmosDB collection.
    * @param blobId the {@link BlobId} for which metadata is replaced.
    * @param fieldName the metadata field to update.
    * @param value the new value for the field.
    * @return the {@link ResourceResponse} returned by the operation, if successful.
    * Returns {@Null} if the document is not found or the field already has the specified value.
```

Comment:
This comment says that null will be returned if the document is not found. I assume this has been changed to throw an exception in this case?